// react-material-alert/lib/Alert.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Natures = require('./Natures');

var _Natures2 = _interopRequireDefault(_Natures);

var _ic_close_18px = require('../images/ic_close_18px.svg');

var _ic_close_18px2 = _interopRequireDefault(_ic_close_18px);

require('../css/main.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Alert = function (_React$Component) {
  (0, _inherits3.default)(Alert, _React$Component);

  function Alert(props) {
    (0, _classCallCheck3.default)(this, Alert);

    var _this = (0, _possibleConstructorReturn3.default)(this, (0, _getPrototypeOf2.default)(Alert).call(this, props));

    _this.state = { sShow: true };
    _this.dismissAlert = _this.dismissAlert.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(Alert, [{
    key: 'dismissAlert',
    value: function dismissAlert() {
      this.setState({ sShow: false });
      if (this.props.closeCallback) {
        this.props.closeCallback();
      }
    }
  }, {
    key: 'getAlertStyle',
    value: function getAlertStyle() {
      var alertStyle = {
        margin: 5 + 'px',
        position: 'relative',
        minHeight: 50 + 'px',
        display: 'flex',
        justifyContent: 'center', /* align horizontal */
        alignItems: 'center', /* align vertical */
        fontFamily: 'roboto'
      };

      var alertWarning = (0, _extends3.default)({}, alertStyle, { backgroundColor: '#FDD835' });
      var alertFailure = (0, _extends3.default)({}, alertStyle, { backgroundColor: '#F44336' });
      var alertInfo = (0, _extends3.default)({}, alertStyle, { backgroundColor: '#BBDEFB' });
      var alertSuccess = (0, _extends3.default)({}, alertStyle, { backgroundColor: '#69F0AE' });

      switch (this.props.alert.nature) {
        case _Natures2.default.WARNING:
          return alertWarning;
        case _Natures2.default.FAILURE:
          return alertFailure;
        case _Natures2.default.SUCCESS:
          return alertSuccess;
        case _Natures2.default.INFO:
          return alertInfo;
      }
    }
  }, {
    key: 'getCloseBtnStyle',
    value: function getCloseBtnStyle() {
      return {
        position: 'absolute',
        right: 0 + 'px',
        top: 0 + 'px',
        cursor: 'pointer'
      };
    }
  }, {
    key: 'render',
    value: function render() {
      if (this.state.sShow === false) {
        return _react2.default.createElement('div', null);
      }
      var Content = this.props.alert.content;
      return _react2.default.createElement(
        'div',
        { style: this.getAlertStyle(), role: 'alert', id: this.props.id },
        _react2.default.createElement('img', { src: _ic_close_18px2.default, onClick: this.dismissAlert, style: this.getCloseBtnStyle() }),
        _react2.default.createElement(Content, null)
      );
    }
  }]);
  return Alert;
}(_react2.default.Component);

Alert.propTypes = {
  alert: _react2.default.PropTypes.shape({
    nature: _react2.default.PropTypes.number.isRequired,
    content: _react2.default.PropTypes.func.isRequired
  })
};
exports.default = Alert;
